homeassistant/core.py

```diff
@@ -252,7 +246,8 @@ def async_block_till_done(self):
 
     def stop(self) -> None:
         """Stop Home Assistant and shuts down all threads."""
-        run_coroutine_threadsafe(self.async_stop(), self.loop)
+        self.loop.call_soon_threadsafe(
+            self.loop.create_task, self.async_stop())
```

Comment:
https://docs.python.org/3/library/asyncio-dev.html#concurrency-and-multithreading
> To schedule a coroutine object from a different thread, the run_coroutine_threadsafe() function should be used.

Also, the tests won't pass if I just call create_task.